### PR TITLE
isight integration

### PR DESCRIPTION
**Pull request by asafshen**
integration is not ready
I would like to get a general review on the commands & context

please note that `HMAC_SHA256_MAC` implementation is here (lines 47-294)
it will be moved to common later

### CODE REVIEW

File: Integrations/integration-isight.yml [hunk 1]
```diff
@@ -0,0 +1,405 @@
+commonfields:
+  id: JS-Isight
+  version: -1
+name: JS-Isight
+display: JS-Isight
+category: Analytics & SIEM
```
Comment (Spivaka) [inline Integrations/integration-isight.yml]: we have PANW icon in wildfire + panorama

File: Integrations/integration-isight.yml [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+  script: |
+    var baseUrl = 'https://api.isightpartners.com'; // iSight base url
+    var publicKey = params.publicKey;
+    var privateKey = params.privateKey;
+    var acceptVersion = params.version;
+    var insecure = params.insecure;
+    var proxy = params.proxy;
+
+    // handle '/' at the end of baseUrl
+    if (baseUrl[baseUrl.length - 1] === '/') {
```
Comment (Spivaka) [inline Integrations/integration-isight.yml]: :(

**Comment by asafshen** [comment]
@anara123 ready

File: Integrations/integration-isight.yml [hunk 3]
```diff
@@ -0,0 +1,446 @@
...
+
+    // we use this to determine the order of reputation sevirity of the types
+    var intelligenceTypeOrder = ['overview', 'vulnerability', 'malware', 'threat'];
+
+    var epoch2DateStr = function(epochStr) {
+        return new Date(parseInt(epochStr) * 1000).toString();
+    }
+
+    var createTableEntry = function (name, contents, table, context) {
+            return {
```
Comment (anara123) [inline Integrations/integration-isight.yml]: fix indent

File: Integrations/integration-isight.yml [hunk 4]
```diff
@@ -0,0 +1,446 @@
...
+            throw 'Request Failed.\nUrl: ' + requestUrl + '\nStatus code: ' + res.StatusCode + '.\nResult: ' + JSON.stringify(res);
+        }
+
+        if (res.StatusCode === 204) {
+            return [];
+        }
+
+        var body = JSON.parse(res.Body);
+        if (!body || !body.success) {
+            throw 'Request Failed.\nResonse body: ' + body;q
```
Comment (anara123) [inline Integrations/integration-isight.yml]: remove `q`

File: Integrations/integration-isight.yml [hunk 5]
```diff
@@ -0,0 +1,446 @@
...
+    var basicSearchIP = function(ip) {
+        var records = basicSearch('ip', ip);
+
+        if (!records) {
+            return "No items match your search";
+        }
+
+        var res = createContextReportsAndScore(records);
+        var context = {
+            DBotScore: {Indicator: ip, Type: 'IP', Vendor: VENDOR_NAME, Score: res.dbotScore}
```
Comment (anara123) [inline Integrations/integration-isight.yml]: missing comma

File: Integrations/integration-isight.yml [hunk 6]
```diff
@@ -0,0 +1,446 @@
...
+    var basicSearchfile = function(key, value) {
+        var records = basicSearch(key, value);
+
+        if (!records) {
+            return "No items match your search";
+        }
+
+        var res = createContextReportsAndScore(records);
+        var context = {
+            DBotScore: {Indicator: value, Type: 'file', Vendor: VENDOR_NAME, Score: res.dbotScore}
```
Comment (anara123) [inline Integrations/integration-isight.yml]: missing comma

File: Integrations/integration-isight.yml [hunk 7]
```diff
@@ -0,0 +1,446 @@
...
+
+        var table = [{
+            ID: report.reportId,
+            title: report.title,
+            intelligenceType: report.intelligenceType,
+            audience: report.audience.join(),
+            publishDate: report.publishDate,
+            ThreatScape: report.ThreatScape.product.join(),
+            operatingSystems: report.operatingSystems,
+            riskRating: report.riskRating
```
Comment (anara123) [inline Integrations/integration-isight.yml]: missing comma

**Comment by asafshen** [comment]
@anara123 thanks!
Fixed review notes

**Pull request merged**